### PR TITLE
fix: update package name

### PR DESCRIPTION
**Pull request by MichaelDeBoey**

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
I seems like ESLint can't find the plugin because we're not adhering [naming conventions](https://eslint.org/docs/latest/user-guide/configuring/plugins#naming-convention) anymore with renaming the package to `@eslint-community/mysticatea__eslint-plugin`.

@eslint-community/eslint-tsc @eslint-community/mysticatea-eslint-plugin Should we rename the package (& repo) to `@eslint-community/eslint-plugin-mysticatea` instead? 🤔

**Comment by MichaelDeBoey** [comment]
I seems like ESLint can't find the plugin because we're not adhering [naming conventions](https://eslint.org/docs/latest/user-guide/configuring/plugins#naming-convention) anymore with renaming the package to `@eslint-community/mysticatea__eslint-plugin`.

@eslint-community/eslint-tsc @eslint-community/core-team @eslint-community/mysticatea-eslint-plugin Should we rename the package (& repo) to `@eslint-community/eslint-plugin-mysticatea` instead? 🤔

**Comment by nzakas** [comment]
> Should we rename the package (& repo) to @eslint-community/eslint-plugin-mysticatea instead? 🤔

Yes. For plugins, names must begin with `eslint-plugin-`.

File: README.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-[![codecov](https://codecov.io/gh/mysticatea/eslint-plugin/branch/master/graph/badge.svg)](https://codecov.io/gh/mysticatea/eslint-plugin)
-[![Dependency Status](https://david-dm.org/mysticatea/eslint-plugin.svg)](https://david-dm.org/mysticatea/eslint-plugin)
+[![npm version](https://img.shields.io/npm/v/@eslint-community/eslint-plugin-mysticatea.svg)](https://www.npmjs.com/package/@eslint-community/eslint-plugin-mysticatea)
+[![Downloads/month](https://img.shields.io/npm/dm/@eslint-community/eslint-plugin-mysticatea.svg)](http://www.npmtrends.com/@eslint-community/eslint-plugin-mysticatea)
+[![Build Status](https://github.com/eslint-community/mysticatea__eslint-plugin/workflows/CI/badge.svg)](https://github.com/eslint-community/mysticatea__eslint-plugin/actions)
+[![codecov](https://codecov.io/gh/eslint-community/mysticatea__eslint-plugin/branch/master/graph/badge.svg)](https://codecov.io/gh/eslint-community/mysticatea__eslint-plugin)
+[![Dependency Status](https://david-dm.org/eslint-community/mysticatea__eslint-plugin.svg)](https://david-dm.org/eslint-community/mysticatea__eslint-plugin)
```
Comment (ota-meshi) [inline README.md]:
```suggestion
[![Build Status](https://github.com/eslint-community/eslint-plugin-mysticatea/workflows/CI/badge.svg)](https://github.com/eslint-community/eslint-plugin-mysticatea/actions)
[![codecov](https://codecov.io/gh/eslint-community/eslint-plugin-mysticatea/branch/master/graph/badge.svg)](https://codecov.io/gh/eslint-community/eslint-plugin-mysticatea)
[![Dependency Status](https://david-dm.org/eslint-community/eslint-plugin-mysticatea.svg)](https://david-dm.org/eslint-community/eslint-plugin-mysticatea)
```

File: README.md [hunk 2]
```diff
@@ -135,7 +135,7 @@ This plugin follows [semantic versioning](http://semver.org/) and [ESLint's Sema
 
 ## 📰 Changelog
 
-- [GitHub Releases](https://github.com/mysticatea/eslint-plugin/releases)
+- [GitHub Releases](https://github.com/eslint-community/mysticatea__eslint-plugin/releases)
```
Comment (ota-meshi) [inline README.md]:
```suggestion
- [GitHub Releases](https://github.com/eslint-community/eslint-plugin-mysticatea/releases)
```

File: package.json [hunk 3]
```diff
@@ -50,7 +50,7 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/mysticatea/eslint-plugin.git"
+    "url": "https://github.com/eslint-community/mysticatea__eslint-plugin"
```
Comment (ota-meshi) [inline package.json]:
```suggestion
    "url": "https://github.com/eslint-community/eslint-plugin-mysticatea"
```

File: package.json [hunk 4]
```diff
@@ -68,7 +68,7 @@
   "author": "Toru Nagashima",
   "license": "MIT",
   "bugs": {
-    "url": "https://github.com/mysticatea/eslint-plugin/issues"
+    "url": "https://github.com/eslint-community/mysticatea__eslint-plugin/issues"
```
Comment (ota-meshi) [inline package.json]:
```suggestion
    "url": "https://github.com/eslint-community/eslint-plugin-mysticatea/issues"
```

File: package.json [hunk 5]
```diff
@@ -68,7 +68,7 @@
...
   },
-  "homepage": "https://github.com/mysticatea/eslint-plugin#readme"
+  "homepage": "https://github.com/eslint-community/mysticatea__eslint-plugin#readme"
```
Comment (ota-meshi) [inline package.json]:
```suggestion
  "homepage": "https://github.com/eslint-community/eslint-plugin-mysticatea#readme"
```

File: lib/rules/arrow-parens.js [hunk 6]
```diff
@@ -31,7 +31,7 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false,
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/arrow-parens.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/arrow-parens.md",
```
Comment (ota-meshi) [inline lib/rules/arrow-parens.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/arrow-parens.md",
```

File: lib/rules/block-scoped-var.js [hunk 7]
```diff
@@ -197,7 +197,7 @@ module.exports = {
             category: "Possible Errors",
             recommended: false,
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/block-scoped-var.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/block-scoped-var.md",
```
Comment (ota-meshi) [inline lib/rules/block-scoped-var.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/block-scoped-var.md",
```

File: lib/rules/no-instanceof-array.js [hunk 8]
```diff
@@ -15,7 +15,7 @@ module.exports = {
             description: "disallow 'instanceof' for Array",
             category: "Best Practices",
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/no-instanceof-array.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/no-instanceof-array.md",
```
Comment (ota-meshi) [inline lib/rules/no-instanceof-array.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/no-instanceof-array.md",
```

File: lib/rules/no-instanceof-wrapper.js [hunk 9]
```diff
@@ -15,7 +15,7 @@ module.exports = {
             description: "disallow 'instanceof' for wrapper objects",
             category: "Best Practices",
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/no-instanceof-wrapper.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/no-instanceof-wrapper.md",
```
Comment (ota-meshi) [inline lib/rules/no-instanceof-wrapper.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/no-instanceof-wrapper.md",
```

File: lib/rules/no-literal-call.js [hunk 10]
```diff
@@ -22,7 +22,7 @@ module.exports = {
             category: "Possible Errors",
             recommended: false,
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/no-literal-call.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/no-literal-call.md",
```
Comment (ota-meshi) [inline lib/rules/no-literal-call.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/no-literal-call.md",
```

File: lib/rules/no-this-in-static.js [hunk 11]
```diff
@@ -15,7 +15,7 @@ module.exports = {
             description: "Disallow `this`/`super` in static methods",
             category: "Best Practices",
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/no-this-in-static.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/no-this-in-static.md",
```
Comment (ota-meshi) [inline lib/rules/no-this-in-static.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/no-this-in-static.md",
```

File: lib/rules/no-use-ignored-vars.js [hunk 12]
```diff
@@ -21,7 +21,7 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false,
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/no-use-ignored-vars.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/no-use-ignored-vars.md",
```
Comment (ota-meshi) [inline lib/rules/no-use-ignored-vars.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/no-use-ignored-vars.md",
```

File: lib/rules/no-useless-rest-spread.js [hunk 13]
```diff
@@ -98,7 +98,7 @@ module.exports = {
             category: "Best Practices",
             recommended: false,
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/no-useless-rest-spread.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/no-useless-rest-spread.md",
```
Comment (ota-meshi) [inline lib/rules/no-useless-rest-spread.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/no-useless-rest-spread.md",
```

File: lib/rules/prefer-for-of.js [hunk 14]
```diff
@@ -498,7 +498,7 @@ module.exports = {
             category: "Best Practices",
             recommended: false,
             url:
-                "https://github.com/mysticatea/eslint-plugin/blob/v13.0.0/docs/rules/prefer-for-of.md",
+                "https://github.com/eslint-community/mysticatea__eslint-plugin/blob/v13.0.0/docs/rules/prefer-for-of.md",
```
Comment (ota-meshi) [inline lib/rules/prefer-for-of.js]:
```suggestion
                "https://github.com/eslint-community/eslint-plugin-mysticatea/blob/v13.0.0/docs/rules/prefer-for-of.md",
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

**Review by ota-meshi — APPROVED** [review]
Looks good to me! Thank you!

**Pull request merged**